### DISCUSSION TITLE
magit + zenburn highlighting very hard to read

### DISCUSSION THREAD

**Discussion by andyhhp** [magit/magit#5150] [Support] [answered]
I have just updated Debian from bullseye to bookworm.  This has changed the colours in magit.

The package versions are now:
```
$ apt-cache policy emacs elpa-zenburn-theme elpa-magit
emacs:
  Installed: 1:28.2+1-15
  Candidate: 1:28.2+1-15
  Version table:
 *** 1:28.2+1-15 500
        500 http://deb.debian.org/debian bookworm/main amd64 Packages
        100 /var/lib/dpkg/status
elpa-zenburn-theme:
  Installed: 2.7.0-2
  Candidate: 2.7.0-2
  Version table:
 *** 2.7.0-2 500
        500 http://deb.debian.org/debian bookworm/main amd64 Packages
        100 /var/lib/dpkg/status
elpa-magit:
  Installed: 3.3.0-2
  Candidate: 3.3.0-2
  Version table:
 *** 3.3.0-2 500
        500 http://deb.debian.org/debian bookworm/main amd64 Packages
        100 /var/lib/dpkg/status
```
and no local font/face customisations.

By and large, the new colours are fine, except for the in-focus section, where the additional highlighting makes the result especially hard to read:
![image](https://github.com/magit/magit/assets/697621/5e00b2bc-a8c9-496e-8cf2-8082ff16bd38)

The shade of green background changing is the worst problem, but the red getting brighter isn't great either.

If I disable zenburn theme, then same magit-revision view only changes the background of the context, taking it from transparent to grey, but leaves the added/removed part of the hunks rendered the same whether they're in focus or not.  (I do quite like this mechanic.)

However, with zenburn the background of the added/removed regions change too.

I'm afraid that I don't know enough to know whether this is a magit issue or a zenburn issue.  Can anyone advise?

**Comment by tarsius** [comment]
I've taken a quick look at `zenburn-theme.el` and the last changes to the relevant areas (the `diff` and `magit > headings and diffs` sections) were modified five years ago. The last relevant changes in Magit are even older.

Maybe you just updated from an extremely old version to an old or new version. [Update: The output you posted indicates that neither zenburn nor magit got updated (versions stayed the same). So it seems unlikely that this is due to a "change" in either of these two packages.]

I don't recall how exactly this is supposed to look in this theme. It appears to me too low-contrast too.

I am afraid I cannot help you. Maybe you have more luck on the emacs reddit. Good luck!

As always, have you tried turning it of and on again?

**Reply by andyhhp** [reply]
I have very much turned it off and on again... I hit Debian Bug 1040690/1040787 doing this update and was forced to uninstall/reinstall all elpa packages to clean up.

According to the dpkg log, I had these versions changes:
```
upgrade elpa-zenburn-theme:all 2.6-3 2.7.0-2
upgrade elpa-magit:all 2.99.0.git0957.ge8c7bd03-1 3.3.0-2
```
So it's quite possible I did go from something very old to something reasonably new.

Testing on a different Ubuntu 22.04 I have to hand (same magit/zenburn packages as Bookworm), an observation is that the issue is less pronounced in a proper emacs window.  The first screenshot was an ssh terminal to a headless server.  Here are the two side-by-side:

![image](https://github.com/magit/magit/assets/697621/089ea540-3aff-416e-89bd-84dff76bb2b0)

Presumably given that zenburn has overrides for all magit faces, it's zenburn that needs playing with to add a smidgin more contrast?

**Reply by tarsius** [reply, marked as answer]
Terminals can't always display all the colors that a gui can display. Some can (i.e., have to) be configured to do so. If they are told to use a color that they think they cannot display, then they display a similar color. This appears to be what is happening here.

https://github.com/termstandard/colors

**Reply by andyhhp** [reply]
Oh - excellent link.  `COLORTERM=truecolor` in the environment seems to fix this.

That's definitely less messy than trying to customise the faces.

Thankyou for your help, and as this is my first time here, thank-you for an excellent git interface
